examples/waspleau/ext/serverSetup.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Setup Function
+ * 
+ * This function is what we default export from this module.
+ * It's primary duty is to setup the workers functions, the queue,
+ * and kick off the jobs that will invoke worker functions.
+ * */
+export default async () => {
+  setupWorkerFunctions()
+  await setupQueue()
+  await populateQueue()
+}
+
+
+/**
+ * Dashboard Data
+ * 
+ * This is our in-memory data store we send down to the client via a query in dashboard.js.
+ * This of course means every time the server is restarted we lose this data and must repopulate.
+ * */
+const dashboardData = {}
+export const getDashboardData = () => Object.values(dashboardData).flat()
+
+
+/** 
+ * Workers
+ * 
+ * Below are two sample workers. The interface they are expected to conform to are:
+ * (1) export a named worker object that has the following shape:
+ *    { name: '<must be unique for all workers>', fn: async function(opts) {}, schedule: '<cron syntax>' }
+ * (2) the async worker function `fn` should return data to save in-memory and display
+ *     in the dashboard of the following shape:
+ *    [{ name: '', value: '', updatedAt: '' }, ...]
+ * 
+ * We use the worker name as the lookup key for the function to run when we process jobs.
+ * The returned data is an array, so any given worker can have multiple tiles of data in the dashboard.
+ * */
+import { githubWorker } from './workers/github.js'
+import { loadTimeWorker } from './workers/loadTime.js'
+
+const workers = [githubWorker, loadTimeWorker]
+const workerFunctions = {}
+
+const setupWorkerFunctions = () => {
+  for (const worker of workers) {
+    workerFunctions[worker.name] = worker.fn
+  }
+}
+
+
+/**
+ * Queue
+ * 
+ * Here we setup the Redis-backed Bull queue named 'waspleau'.
+ * Whenever we process a job in the queue's callback function, we look up the worker name and
+ * call the associated worker function. We then store the returned data to send to the client.
+ * 
+ * There is also a MOCK_DATA env var we check for and pass as an opt in case the worker
+ * wants to send mock data while doing active development.
+ * 
+ * Lastly, we populate the queue with a one-off job on startup for each worker, while also
+ * schedulign the repeated jobs using cron syntax.
+ * */
+import Queue from 'bull'
+
+const queue = new Queue('waspleau', process.env.REDIS_URL || 'redis://127.0.0.1:6379')
+
+const setupQueue = async () => {
+  // Clear entire queue when starting up.
+  // Warning: this is not cluster friendly nor a production best practice!
+  // This just helps keep our Redis queue tidy after lots of jobs/restarts.
```

Comment:
What is the good production practice then, how would we keep the queue tidy in that case?
Btw maybe you could check here for ENV that is being used, and then use that to avoid using this in production?

Reply:
Agree the comment here is not complete nor ideal. I was trying to convey in dev mode it is helpful (so we could wrap in an if), but it is also helpful and _probably_ ok in prod, but not really recommended by anyone. The two issues it solves are:
(1) addressed in comment, but removes old jobs. This can be done other ways though: https://github.com/OptimalBits/bull/issues/840 I am new to Bull and just didn't see that option before so I can use it and remove the comment!
(2) the bigger issue is that any time the recurring job name or cron schedule change, the old job is _not_ removed. So they could have multiple jobs running for the same logical one. So I could just have a note in there about that behavior and say whenever you do change either a job name or cron schedule to run this for safety (protected by an ENV VAR).

There are ways you can address (2) as well: https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#queueremoverepeatable The hard part is I cannot really do that step for them since I won't know what the old job name was, only they will, I would have to remove them all anyways to provide actual safety.

Reply:
Updated here: https://github.com/wasp-lang/wasp/pull/412/commits/46cfc71a756ca2ac240d416d4ace85162b11f92a

Reply:
Makes sense! It seems to me like it would be ok even for production, in our case, to clean the queue, since we don't really care much if an old job doesn't finish, as long as there is a new job coming after it, right? So clearing entire queue when starting up doesn't sound so bad to me. Ok, it depends on the type of workers I guess, but at the moment it is ok. Ok, no need to spend much time here / over-engineer, I just wanted to get some clarity on the topic, thanks!

Reply:
Yep no sweat, and I agree. Even in prod I think this should be fine, but yes there is the possibility of a race condition I think if you have it setup in a clustered setup where someone could start accepting while another boots and clears queue. I can make that final note in there, but agree, I didn't want to spend the time getting it to a 100% bulletproof solution to this corner case. Thanks